Sources/SwiftCrossUI/App.swift

```diff
@@ -119,48 +124,33 @@ extension App {
     }
 
     private static func extractMetadataAndInitializeLogging() {
-        // set up a temporary logger for `extractSwiftBundlerMetadata()` -- we
-        // won't initialize the actual logger until after that returns, so that
-        // users can use the app metadata in a custom logger
-        let temporaryLogHandler = InMemoryLogHandler()
-        _logger = Logger(
-            label: "SwiftCrossUI",
-            factory: { _ in temporaryLogHandler }
-        )
-
-        swiftBundlerAppMetadata = extractSwiftBundlerMetadata()
+        // extract metadata _before_ initializing the logger, so users can use
+        // said metadata when declaring a custom logger
+        let result = Result {
+            swiftBundlerAppMetadata = try extractSwiftBundlerMetadata()
+        }
 
-        // now initialize the real thing...
         _logger = Logger(
             label: "SwiftCrossUI",
             factory: logHandler(label:metadataProvider:)
         )
 
-        // ...and print out any log entries
-        for entry in temporaryLogHandler.entries {
-            logger.log(
-                level: entry.level,
-                entry.message,
-                metadata: entry.metadata
+        // check for an error once the logger is ready
+        if case .failure(let error) = result {
+            logger.error(
+                "failed to extract swift-bundler metadata",
+                metadata: ["error": "\(error)"]
```

Comment:
Does `"\(error)"` correctly use the error's localized description here? Or does it do the useless default description where it's like `SwiftBundlerMetadataError 1 operation failed` or something?